systemd/gssproxy.service.in

```diff
@@ -5,30 +5,50 @@ After=syslog.target network.target
 Before=rpc-gssd.service
 
 [Service]
-StateDirectory=gssproxy/clients gssproxy/rcache
+ConfigurationDirectory=gssproxy
+StateDirectory=gssproxy gssproxy/clients gssproxy/rcache
 Environment=KRB5RCACHEDIR=/var/lib/gssproxy/rcache
 ExecStart=@sbindir@/gssproxy -i
 # This can be changed to notify-reload and ExecReload= can be removed once
 #   systemd 253 is common enough
 Type=notify
 ExecReload=/bin/kill -HUP $MAINPID
 
-ProtectSystem=full
-ProtectClock=true
-ProtectKernelLogs=true
-ProtectControlGroups=true
-RestrictRealtime=true
-# NoNewPrivileges: when true breaks the ability to open a socket
-#   under /var/lib/gssproxy so no NoNewPrivileges
-# PrivateTmp: can't be used as it hides ccaches stored in /tmp
-# ProtectHome: blocks access to /home which may hold ccaches
-# ProtectHostname: blocks propagation of hostname on change
-#   but in some cases, when using a keytab, we may want to see hostname
-#   changes as the server will want to respond only for the system name
-# ProtectKernelTunables: blocks ability to write to proc.
-#   on startup gssproxy needs to write in proc to let nfsd know it can
-#   use the "new" gssproxy method instead of the old rpc stuff.
-
+ProtectSystem=strict
+# Hides ccaches stored in /tmp
+PrivateTmp=no
+PrivateDevices=yes
+PrivateNetwork=yes
+PrivateIPC=yes
+# Blocks access to /home which may hold ccaches
+PrivateUsers=no
+ProtectHome=read-only
```

Comment:
This is wrong, caches may need to be writable.

Reply:
Ok, I wasn't sure about that one, I'll fix that

Reply:
`ProtectSystem=strict` already makes the whole FS read-only (except `/var/lib/gsssproxy` which is made RW via the `StateDirectory=` stanza)...so, we'd need to create `BindPaths=` exceptions for any location where writeable ccaches may be expected....when/where/under which circumstances are there KRB5 ccaches under `/home` and/or `/root`? All systems I've used have ccaches under `/tmp`...(unless the kernel keyring or something like `sssd`s KCM are used)...I need some pointers

Reply:
System specific, the traditional /tmp is considerd broken due TokTou, so on some systems people use their home directories. But there is also the simple case of a user creating a ccache anywhere by just setting KRB5CCNAME in the process (and adding a corresponding gssproxy configuration stanza to point to it).
The DIR type at some point pointed to /run/user directories.
The Keyring or KCM cases are the easy one to handle (guess why I add both those facilities in krb5 and SSSD :-) 

I do not think I can change this now, it would break any user that decided to place writable cacches in a non standard place.

Reply:
Gotcha, so your concern is systems with customised ccache paths via e.g. `KRB5CCNAME` or a customised `/etc/krb5.conf` along these lines:

```
[libdefaults]
default_cc_name = FILE:${homedir}/.krb5cc
```

So which paths would you like to see as being made read-writeable by default (note, it can still be changed locally via systemd drop-ins)? The whole `/root`, `/home` and `/run/user` hierarchy? Or could we default to a more restricted subset and document that systems with such changes should also create a suitable drop-in...?

Reply:
I think the regular paths a user can write in should be allowed, so /home /root /run/user sounds reasonable.
And yes the rest via drop-in.